### ISSUE TITLE
pvc -> pv mismatch when bindng

### ISSUE THREAD

**Issue by pkit** [IBM/core-dump-handler#78]
Now, moving from `dev` to `staging` got into the following issue:
```
$ kubectl get pv core-volume
NAME          CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                        STORAGECLASS   REASON   AGE
core-volume   10Gi       RWO            Retain           Bound    core-dump/host-storage-pvc   hostclass               36m
$ kubectl get pv host-volume
NAME          CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS   REASON   AGE
host-volume   1Gi        RWO            Retain           Available           hostclass               16m
$ kubectl get pvc -n core-dump
NAME               STATUS    VOLUME        CAPACITY   ACCESS MODES   STORAGECLASS   AGE
core-storage-pvc   Pending                                           hostclass      7m40s
host-storage-pvc   Bound     core-volume   10Gi       RWO            hostclass      37m
```
As you can see the `host-storage-pvc` is bound to `core-volume` pv, which makes it impossible for `core-storage-pvc` to bind (1Gi vs 10Gi)
Looks like you either need to pre-bind or use some other creative idea.
Will think about something, but I don't have anything rn.

**Comment by No9** [comment]
That's a nasty one that was missed because the default is to create the `core` directory as a sub-folder of the host directory.
 
Looking at the docs this can be fixed with a reservation using a [claimRef](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reserving-a-persistentvolume).
 
```diff
diff --git a/charts/core-dump-handler/templates/core-storage-pv.yaml b/charts/core-dump-handler/templates/core-storage-pv.yaml
index a89c5a7..c8ed150 100644
--- a/charts/core-dump-handler/templates/core-storage-pv.yaml
+++ b/charts/core-dump-handler/templates/core-storage-pv.yaml
@@ -6,6 +6,9 @@ metadata:
     type: local
 spec:
   storageClassName: {{ .Values.storageClass }}
+  claimRef:
+    name: core-storage-pvc
+    namespace: {{ .Release.Namespace }}
   capacity:
     storage: {{ .Values.coreStorage }}
   accessModes:
diff --git a/charts/core-dump-handler/templates/host-storage-pv.yaml b/charts/core-dump-handler/templates/host-storage-pv.yaml
index 840a4fd..2e69897 100644
--- a/charts/core-dump-handler/templates/host-storage-pv.yaml
+++ b/charts/core-dump-handler/templates/host-storage-pv.yaml
@@ -6,6 +6,9 @@ metadata:
     type: local
 spec:
   storageClassName: {{ .Values.storageClass }}
+  claimRef:
+    name: host-storage-pvc
+    namespace: {{ .Release.Namespace }}
   capacity:
     storage: {{ .Values.hostStorage }}
   accessModes:
 ```

This now gives the intended result
```
$ kubectl get pv core-volume
NAME          CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                      STORAGECLASS   REASON   AGE
core-volume   10Gi       RWO            Retain           Bound    observe/core-storage-pvc   hostclass               6m32s

$ kubectl get pv host-volume
NAME          CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                      STORAGECLASS   REASON   AGE
host-volume   1Gi        RWO            Retain           Bound    observe/host-storage-pvc   hostclass               6m42s

$ kubectl get pvc -n observe
NAME               STATUS   VOLUME        CAPACITY   ACCESS MODES   STORAGECLASS   AGE
core-storage-pvc   Bound    core-volume   10Gi       RWO            hostclass      6m55s
host-storage-pvc   Bound    host-volume   1Gi        RWO            hostclass      6m55s
```
And passes the integration tests.

`.Release.namespace` is picked up by `helm template` so it should work for you.
If your in agreement I'll commit and release.

**Comment by No9** [comment]
Added to this branch https://github.com/IBM/core-dump-handler/tree/claim-ref

**Comment by pkit** [comment]
I think `claimRef` should work, yeah. Thanks!
Will test it later though, doing some different stuff rn.

**Comment by pkit** [comment]
Yup, works fine

**Comment by No9** [comment]
Hey @pkit  
Released in v8.1.1

**Comment by pkit** [comment]
Thanks!
